### DISCUSSION TITLE
Document k3s.io annotations

### DISCUSSION THREAD

**Discussion by aragilar** [k3s-io/k3s#9053] [General]
**Is your feature request related to a problem? Please describe.**
As far as I can see, there's no official list of annotations in the k3s.io namespace. Having such a list (and documenting when to use said annotations) would be really helpful in confirming that a cluster is operating in the correct and expected state.

**Describe the solution you'd like**
As above, a page on https://docs.k3s.io/ with a list of all the official k3s annotations would be really helpful.

**Describe alternatives you've considered**
N/A

**Additional context**
While this is really only asking about k3s.io, I'm aware that rancher and cattle.io exists, so having those documented on the new page (or on a different page, and a link between the two provided) would be really nice!

**Comment by dereknola** [comment]
There isn't any documentation for the k3s.io annotation because they are largely for internal use within K3s itself. A typical set of them is:
```
   "k3s.io/hostname": "my_pc_name",
   "k3s.io/internal-ip": "192.168.1.234,2600:6c52:7a00:9f4:388d:a7cd:2423:4e13",
   "k3s.io/node-args": "[\"server\"]",
   "k3s.io/node-config-hash": "3asdfawsdfL5P5K3DHSJGUCKZMP35JAGEHJNVKYQ====",
   "k3s.io/node-env": "{\"K3S_DATA_DIR\":\"/var/lib/rancher/k3s/data/76b01e0750d8fb96b0b82f1cac8806cd371963
```

Validation of these annotations would be somewhat pointless, all of these are configured via k3s argument flags, or generated from the result of those flags, so validation your configuration file `/etc/rancher/k3s/config.yaml` would be more helpful. None of them should be touched by the user directly.

**Comment by aragilar** [comment]
What about `egress.k3s.io/cluster=true`? I'm not trying to manually set them, instead confirming that k3s is picking up the right information (e.g. in the presence of multiple NICs), and there isn't some CLI flag I should be passing in.

**Comment by caroline-suse-rancher** [comment]
I'm converting this to a discussion, as this is unlikely to be something we implement, but we can circle back to get your questions answered.
